pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/ResultTableResultSet.java

```diff
@@ -61,6 +63,33 @@ public String getString(int rowIndex, int columnIndex) {
     }
   }
 
+  public List<String> getAllColumns() {
+    List<String> columns = new ArrayList<>();
+    if (_columnNamesArray == null) {
+      return columns;
+    }
+
+    for (JsonNode column : _columnNamesArray) {
+      columns.add(column.textValue());
+    }
+
+    return columns;
+  }
+
+
+  public List<String> getAllColumnsDataTypes() {
+    List<String> columnDataTypes = new ArrayList<>();
+    if (_columnDataTypesArray == null) {
```

Comment:
Would it be better to model `  private final JsonNode _columnDataTypesArray;` as a Schema object?

Reply:
It is just an array of strings so does it make sense to create an object for it?
We are anyways not doing it anywhere in java client and using JsonNode.

Reply:
I can create a seperate PR though for replacing all instances of JsonNode with proper classes.